### PR TITLE
Task/14355 add submission unit tests

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Add submission unit tests

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14355

### CODE REVIEW

**Review by flxschmidt — APPROVED** [review]
LGTM! 🍒

**Review by puneetmahali1 — APPROVED** [review]
👍

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForSRS/__tests__/OTPAuthorizationForSRSResourceTests.swift [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+				guard case let .receivedResourceError(customError) = otpError else {
+					XCTFail("unexpected error case")
+					return
+				}
+				XCTAssertEqual(customError, .invalidResponseError)
+			}
+		}
+	}
+	
+	func testGIVEN_AuthorizeOTP_WHEN_NoNetworkConnection_THEN_NetworkErrorReturned() {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForSRS/__tests__/OTPAuthorizationForSRSResourceTests.swift:395]: Can you please add a test case for .deviceBlocked?

**Review by naveeddotio — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForSRS/__tests__/OTPAuthorizationForSRSResourceTests.swift [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+			switch result {
+				// THEN
+			case .success:
+				XCTFail("success should not be called")
+			case .failure(let otpError):
+				guard case let .receivedResourceError(customError) = otpError else {
+					XCTFail("unexpected error case")
+					return
+				}
+				XCTAssertEqual(customError, .otherServerError)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForSRS/__tests__/OTPAuthorizationForSRSResourceTests.swift:333]: Name of the test suggest that it should return internalServerError but in the end we are equating to otherServerError, both of them are different errors if we look into the enum types, am I missing something here?

**Review by naveeddotio — CHANGES_REQUESTED** [review]
Please see comments

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForSRS/__tests__/OTPAuthorizationForSRSResourceTests.swift (hunk 1 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForSRS/__tests__/OTPAuthorizationForSRSResourceTests.swift:395]: Please refer to this PR: https://github.com/corona-warn-app/cwa-app-ios/pull/4928/files

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**